frame-reader.go

```diff
@@ -0,0 +1,150 @@
+// Read a generic "framed" packet consisting of a header and a
+// This is used for both TLS Records and TLS Handshake Messages
+package mint
+
+import (
+	"fmt"
+	"io"
+)
+
+type frameDetails interface {
```

Comment:
Nit: I might prefer if this were named something related to "framing", since you're describing a framing scheme, instead of a frame.  Maybe just call it `framing`?